keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,680 @@
+
+#### Alpha
+
+- Feature implemented behind a feature gate flag option
+- Initial e2e tests completed and enabled
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+
+### Version Skew Strategy
+
+N/A
+
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+To enable this feature requires adding to the Kubelet yaml file for the CPUManager feature gate and add the policy option for uncore cache alignment
+
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+For `CPUManager` it is a requirement going from `none` to `static` policy cannot be done dynamically because of the `cpu_manager_state file`. The node needs to be drained and the policy checkpoint file (`cpu_manager_state`) need to be removed before restarting Kubelet. This feature specifically relies on the `static` policy being enabled.
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `CPUManagerAlphaPolicyOptions`
+  - Components depending on the feature gate: `kubelet`
+- [x] Other
+  - Describe the mechanism: Change the `kubelet` configuration to set a `CPUManager` policy of static then setting the policy option of `align-cpus-by-uncorecache`
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? Yes, a `kubelet` restart is required for changes to take place.
+
+###### Does enabling the feature change any default behavior?
+
+No, to enable this feature, it must be explicitly set in the `CPUManager` static policy and the policy option `align-cpus-by-uncorecache` must be set.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes this feature can be disabled it will just require a restart of `kubelet`. The Kubelet configuration will need to be set with the static policy option and align-cpus-by-uncorecache flag removed.
```

Comment:
I think this could use a note why we don't need a drain (the existing pods will be aligned by uncorecache, but new ones will not be and the behavior will look largely similar)

Reply:
Actually I think this might be wrong because of [static policy](https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/#changing-the-cpu-manager-policy) And says here we do indeed have to drain the node, though the reasoning about offlining CPUs is not clear to me 
@ffromani can you shed some light to this?

Reply:
@ajcaldelas I think @haricommander is referring to rolling back from "static" to non-static.  In which case, I don't think it's necessary to drain the nodes.  CPU allocation is saved in cpu_manager_state file and upon restart, it should assign existing pods to those CPUs and then continue as usual.  I think it's when you move from "non-static" to static state is when kubelet has issue during startup.  Please check.